process/coordinator/process.go

```diff
@@ -1162,6 +1179,147 @@ func getNumOfCrossShardScCallsOrSpecialTxs(
 	return numCrossShardScCallsOrSpecialTxs
 }
 
+// VerifyCreatedMiniBlocks re-checks gas used and generated fees in the given block
+func (tc *transactionCoordinator) VerifyCreatedMiniBlocks(header data.HeaderHandler, body *block.Body) error {
+	if header.GetEpoch() < tc.blockGasAndFeesReCheckEnableEpoch {
+		return nil
+	}
+
+	mapMiniBlockTypeAllTxs := make(map[block.Type]map[string]data.TransactionHandler)
```

Comment:
if the lines L1188-L1194 would have been extracted in a small function, the VerifyCreatedMiniBlocks would have had a perfect structure as all its contents would have been on the same abstraction level.

Reply:
Done